daemon/api.go

```diff
@@ -958,7 +959,34 @@ func changeInterfaces(c *Command, r *http.Request) Response {
 		if len(a.Plugs) == 0 {
 			return BadRequest("at least one plug is required")
 		}
-		err := c.d.interfaces.AddPlug(&a.Plugs[0])
+		// NOTE: This constructs a partial snap.yaml meta-data. Later on it
+		// should reference real meta-data so that it chimes in with real
+		// snaps. Alternatively, this action should be removed.
```

Comment:
I am not sure I understand the below bits (or rather understand why it is done in this way) but I wonder if we could get the real snap metadata here? The daemon has access to it and it seems to me we want this anyway to verify that what the user send us over the REST api matches what we ahave in the app (but maybe that is already done at a different place?).